packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/SerializationTests.kt

```diff
@@ -0,0 +1,649 @@
+                    anyValues.mapIndexed { i, value -> Pair(keys[i].toString(), value) }
+                }
+                else -> throw IllegalArgumentException("Wrong classifier: '$classifier'")
+            } as List<Pair<String, T>>
+        }
+
+        override fun toString(): String = property.name
+
+        override fun getCollection(container: SerializableSample): RealmDictionary<T> =
+            property.get(container)
+
+        override fun createContainerAndGetCollection(realm: MutableRealm): RealmDictionary<T> {
+            val container = SerializableSample().let {
+                realm.copyToRealm(it)
+            }
+            return property.get(container)
+                .also { dictionary ->
+                    assertNotNull(dictionary)
+                    assertTrue(dictionary.isEmpty())
+                }
+        }
+
+        override fun createPrePopulatedContainer(): SerializableSample {
+            return SerializableSample().also {
+                property.get(it)
+                    .apply {
+                        putAll(dataSetToLoad)
+                    }
+            }
+        }
+    }
+
+    private fun <T> KClassifier.assertValue(
+        expected: T,
+        actual: T
+    ) {
+        when (this) {
+            ByteArray::class -> assertContentEquals(expected as ByteArray?, actual as ByteArray?)
+            RealmObject::class -> assertEquals(
+                (expected as SerializableSample).stringField,
+                (actual as SerializableSample).stringField
+            )
+            RealmAny::class -> {
+                expected as RealmAny?
+                actual as RealmAny?
+
+                if (expected != null && actual != null) {
+                    when (expected.type) {
+                        RealmAny.Type.OBJECT -> {
+                            assertEquals(expected.type, actual.type)
+                            assertEquals(
+                                expected.asRealmObject<SerializableSample>().stringField,
+                                actual.asRealmObject<SerializableSample>().stringField
+                            )
+                        }
+                        else -> assertEquals(expected, actual)
+                    }
+                } else if (expected != null || actual != null) {
+                    fail("One of the RealmAny values is null, expected = $expected, actual = $actual")
+                }
+            }
+            else -> assertEquals(expected, actual)
+        }
+    }
+
+    @Test
+    fun exhaustiveRealmTypesTester() {
+        val expected = SerializableSample().apply {
+            nullableObject = SerializableSample()
+            realmEmbeddedObject = SerializableEmbeddedObject()
+        }
+        val encoded: String = json.encodeToString(expected)
+        val decoded: SerializableSample = json.decodeFromString(encoded)
+
+        TypeDescriptor.elementClassifiers.filterNot {
+            arrayOf(
```

Comment:
I didn't add it because it is only used once, but now that you say it, I could add a field in the TypeDescriptor that defines if the type is a Realm data type. Then here we could just apply a filter.